src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelDiscoveryDataAggregator.cs

```diff
@@ -90,36 +77,32 @@ public ParallelDiscoveryDataAggregator()
     /// Aggregate discovery data
     /// Must be thread-safe as this is expected to be called by parallel managers
     /// </summary>
-    public void Aggregate(
-        long totalTests,
-        bool isAborted,
-        Dictionary<string, HashSet<string>> discoveredExtensions)
+    public void Aggregate(DiscoveryCompleteEventArgs discoveryCompleteEventArgs)
     {
         lock (_dataUpdateSyncObject)
         {
-            IsAborted = IsAborted || isAborted;
+            IsAborted = IsAborted || discoveryCompleteEventArgs.IsAborted;
 
-            if (IsAborted)
-            {
-                // Do not aggregate tests count if test discovery is aborted. It is mandated by
-                // platform that tests count is negative for discovery abort event.
-                // See `DiscoveryCompleteEventArgs`.
-                TotalTests = -1;
-                return;
-            }
-
-            TotalTests += totalTests;
+            // Do not aggregate tests count if test discovery is aborted. It is mandated by
+            // platform that tests count is negative for discovery abort event.
+            // See `DiscoveryCompleteEventArgs`.
+            TotalTests = IsAborted ? -1 : TotalTests + discoveryCompleteEventArgs.TotalCount;
 
             // Aggregate the discovered extensions.
-            DiscoveredExtensions = TestExtensions.CreateMergedDictionary(DiscoveredExtensions, discoveredExtensions);
+            DiscoveredExtensions = TestExtensions.CreateMergedDictionary(DiscoveredExtensions, discoveryCompleteEventArgs.DiscoveredExtensions);
         }
+
+        AggregateMetrics(discoveryCompleteEventArgs.Metrics);
+
+        // Do not aggregate NotDiscovered, PartiallyDiscovered, FullyDiscovered sources here
+        // because this aggregator is shared with proxies so the state is already up-to-date.
```

Comment:
```suggestion
        // because this aggregator is shared with proxies so the state is already up-to-date.
        //
        // The reason to share the aggregator is that we only get 1 notification for completed discovery
        // when discovery is done Instead of 1 notification per discovery manager. And we also don't get 
        // any notification for discovery of sources that did not start discovering yet. So we need the state
        // in the aggregator to start with all the sources that we were requested to discover in NotDiscovered
        // state. And then as we discover them, we need to update that directly. So when any discovery is cancelled
        // we can just take the current state in the aggregator and report it.
```